src/components/cursor-controller.js

```diff
@@ -2,451 +2,190 @@ const TARGET_TYPE_NONE = 1;
-const virtualJoystickCutoff = 0.8;
 
-/**
- * Controls virtual cursor behavior in various modalities to affect teleportation, interatables and UI.
- * @namespace user-input
- * @component cursor-controller
- */
 AFRAME.registerComponent("cursor-controller", {
   dependencies: ["raycaster", "line"],
   schema: {
     cursor: { type: "selector" },
     camera: { type: "selector" },
-    playerRig: { type: "selector" },
-    gazeTeleportControls: { type: "selector" },
-    physicalHandSelector: { type: "string" },
-    handedness: { default: "right", oneOf: ["right", "left"] },
     maxDistance: { default: 3 },
-    minDistance: { default: 0.5 },
+    minDistance: { default: 0 },
     cursorColorHovered: { default: "#2F80ED" },
     cursorColorUnhovered: { default: "#FFFFFF" },
-    primaryDown: { default: "action_primary_down" },
-    primaryUp: { default: "action_primary_up" },
-    grabEvent: { default: "action_grab" },
-    releaseEvent: { default: "action_release" }
+    rayObject: { type: "selector" },
+    useMousePos: { default: true },
+    drawLine: { default: false }
   },
 
   init: function() {
+    this.enabled = true;
     this.inVR = false;
     this.isMobile = AFRAME.utils.device.isMobile();
-    this.hasPointingDevice = false;
     this.currentTargetType = TARGET_TYPE_NONE;
-    this.grabStarting = false;
     this.currentDistance = this.data.maxDistance;
     this.currentDistanceMod = 0;
     this.mousePos = new THREE.Vector2();
-    this.controller = null;
-    this.controllerQueue = [];
     this.wasCursorHovered = false;
-    this.wasPhysicalHandGrabbing = false;
     this.origin = new THREE.Vector3();
     this.direction = new THREE.Vector3();
     this.controllerQuaternion = new THREE.Quaternion();
-    this.activeTouch = null;
 
     this.data.cursor.setAttribute("material", { color: this.data.cursorColorUnhovered });
 
-    this._handleTouchStart = this._handleTouchStart.bind(this);
-    this._handleTouchMove = this._handleTouchMove.bind(this);
-    this._handleTouchEnd = this._handleTouchEnd.bind(this);
-    this._handleMouseDown = this._handleMouseDown.bind(this);
-    this._handleMouseMove = this._handleMouseMove.bind(this);
-    this._handleMouseUp = this._handleMouseUp.bind(this);
-    this._handleWheel = this._handleWheel.bind(this);
-    this._handleEnterVR = this._handleEnterVR.bind(this);
-    this._handleExitVR = this._handleExitVR.bind(this);
-    this._handlePrimaryDown = this._handlePrimaryDown.bind(this);
-    this._handlePrimaryUp = this._handlePrimaryUp.bind(this);
-    this._handleModelLoaded = this._handleModelLoaded.bind(this);
-    this._handleCursorLoaded = this._handleCursorLoaded.bind(this);
-    this._handleControllerConnected = this._handleControllerConnected.bind(this);
-    this._handleControllerDisconnected = this._handleControllerDisconnected.bind(this);
+    this.forceCursorUpdate = this.forceCursorUpdate.bind(this);
```

Comment:
I usually only bother binding things if I need to use the bound function for some reason in the same class. If another class needs the bound version it should be responsible for binding it to an instance. And if nothing needs the bound version (which seems to be the case here) then nothing binds it.

Reply:
👍